frontend/src/components/CuisineFilter.tsx

```diff
@@ -0,0 +1,94 @@
+import { cuisineList } from '@/config/restaurant-options-config';
+import { Label } from './ui/label';
+import { Check, ChevronDown, ChevronUp } from 'lucide-react';
+import { ChangeEvent } from 'react';
+import { Button } from './ui/button';
+
+type Props = {
+  onChange(cuisines: string[]): void;
+  selectedCuisines: string[];
+  isExpanded: boolean;
+  onExpandClick(): void;
+};
+
+const CuisineFilter = ({
+  onChange,
+  selectedCuisines,
+  isExpanded,
+  onExpandClick,
+}: Props) => {
+  const handleCuisinesReset = () => {
+    onChange([]);
+  };
+
+  const handleCuisinesChange = (event: ChangeEvent<HTMLInputElement>) => {
+    const clickedCuisine = event.target.value;
+    const isChecked = event.target.checked;
+    const newCuisinesLint = isChecked
+      ? [...selectedCuisines, clickedCuisine]
+      : selectedCuisines.filter(
+          (cuisine: string) => cuisine !== clickedCuisine
+        );
+
+    onChange(newCuisinesLint);
+  };
+
+  return (
+    <>
+      <div className="flex justify-between items-center px-2">
+        <div className="text-md font-semibold mb-2">Filter by Cuisine</div>
+        <div
+          onClick={handleCuisinesReset}
+          className="text-md font-semibold mb-2 underline cursor-pointer text-blue-500"
+        >
+          Rest Filters
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/omkardongre/Food-Ordering-App/pull/41#issuecomment-2407615424>#dd91ac</a><span> - 10/11/2024, 03:13 pm</span>
<br>

<div><small>🟠</small> Medium importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> There's a typo in the 'Reset Filters' button text. It currently says 'Rest Filters' instead of 'Reset Filters'.
</div>
<div id="fix">
<b>Fix:</b> Change the text 'Rest Filters' to 'Reset Filters' to correctly describe the button's function.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -41,7 +41,7 @@
            onClick={handleCuisinesReset}
            className="text-md font-semibold mb-2 underline cursor-pointer text-blue-500"
          >
 -          Rest Filters
 +          Reset Filters
          </div>
        </div>
 
```
</div>
</details>
</div>
<br>


</div>

Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged